src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,49 @@
+package mate.academy.dao;
+
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            movie = (Movie) session.save(movie);
```

Comment:
```suggestion
            session.save(movie);
```

it sets id automatically

Reply:
i don't understand what shout i do here